File: OpenBurnBarCore/Sources/OpenBurnBarCore/Views/Substrate/Families/VolumetricFamily.swift
import Foundation

/// Volumetric family substrate registry. Ported from imaginethat
/// `glyph/stage/styles/volumetric/`.
enum VolumetricFamily {
    static let styles: [SubstrateDescriptor] = [
        SubstrateDescriptor(
            id: "volumetric.sunshaft", family: .volumetric,
            label: "Crepuscular Shafts", hint: "godrays",
            accent: RGBA(r: 120/255, g: 180/255, b: 1.0), accent2: RGBA(r: 1.0, g: 200/255, b: 150/255),
            make: { SunshaftSubstrate() }),
        SubstrateDescriptor(
            id: "volumetric.smoked-glass", family: .volumetric,
            label: "Smoked Glass Slab", hint: "glass",
            accent: RGBA(r: 150/255, g: 170/255, b: 200/255), accent2: RGBA(r: 200/255, g: 210/255, b: 230/255),
            make: { SmokedGlassSubstrate() }),
        SubstrateDescriptor(
            id: "volumetric.silk-filament", family: .volumetric,
            label: "Silk Filament", hint: "filament",
            accent: RGBA(r: 140/255, g: 190/255, b: 1.0), accent2: RGBA(r: 1.0, g: 215/255, b: 170/255),
            make: { SilkFilamentSubstrate() }),
        SubstrateDescriptor(
            id: "volumetric.dust-motes", family: .volumetric,
            label: "Dust Motes", hint: "motes",
            accent: RGBA(r: 200/255, g: 200/255, b: 220/255), accent2: RGBA(r: 1.0, g: 210/255, b: 180/255),
            make: { DustMotesSubstrate() }),
    ]
}

File: windows/particles/OpenBurnBar.Particles/Substrates/Families/VolumetricFamily.cs
using OpenBurnBar.Particles.Model;

namespace OpenBurnBar.Particles.Substrates.Families;

/// <summary>
/// Volumetric family registry — C# port of Swift <c>Families/VolumetricFamily.swift</c>.
/// Depth / cloud-density fields with blur: Crepuscular Shafts + Smoked Glass Slab +
/// Silk Filament + Dust Motes. Ids / labels / accents mirror the Swift descriptors
/// byte-for-byte.
/// </summary>
public static class VolumetricFamily
{
    public static readonly SubstrateDescriptor[] Styles =
    {
        new SubstrateDescriptor(
            id: "volumetric.sunshaft", family: SubstrateFamily.Volumetric,
            label: "Crepuscular Shafts", hint: "godrays",
            accent: new Rgba(120 / 255.0, 180 / 255.0, 1.0),
            accent2: new Rgba(1.0, 200 / 255.0, 150 / 255.0),
            make: () => new SunshaftSubstrate()),
        new SubstrateDescriptor(
            id: "volumetric.smoked-glass", family: SubstrateFamily.Volumetric,
            label: "Smoked Glass Slab", hint: "glass",
            accent: new Rgba(150 / 255.0, 170 / 255.0, 200 / 255.0),
            accent2: new Rgba(200 / 255.0, 210 / 255.0, 230 / 255.0),
            make: () => new SmokedGlassSubstrate()),
        new SubstrateDescriptor(
            id: "volumetric.silk-filament", family: SubstrateFamily.Volumetric,
            label: "Silk Filament", hint: "filament",
            accent: new Rgba(140 / 255.0, 190 / 255.0, 1.0),
            accent2: new Rgba(1.0, 215 / 255.0, 170 / 255.0),
            make: () => new SilkFilamentSubstrate()),
        new SubstrateDescriptor(
            id: "volumetric.dust-motes", family: SubstrateFamily.Volumetric,
            label: "Dust Motes", hint: "motes",
            accent: new Rgba(200 / 255.0, 200 / 255.0, 220 / 255.0),
            accent2: new Rgba(1.0, 210 / 255.0, 180 / 255.0),
            make: () => new DustMotesSubstrate()),
    };
}
